core/src/main/java/com/amplitude/core/utilities/EventsFileManager.kt

```diff
@@ -122,50 +146,96 @@ class EventsFileManager(
         val firstHalfFile = File(directory, "$fileName-1.tmp")
         val secondHalfFile = File(directory, "$fileName-2.tmp")
         val splitStrings = events.split()
-        writeToFile(splitStrings.first, firstHalfFile)
-        writeToFile(splitStrings.second, secondHalfFile)
+        writeEventsToSplitFile(splitStrings.first, firstHalfFile)
+        writeEventsToSplitFile(splitStrings.second, secondHalfFile)
         this.remove(filePath)
     }
 
-    suspend fun getEventString(filePath: String): String = readMutex.withLock {
-        // Block one time of file reads if another task has read the content of this file
-        if (filePathSet.contains(filePath)) {
-            filePathSet.remove(filePath)
-            return ""
-        }
-        filePathSet.add(filePath)
-        File(filePath).bufferedReader().use<BufferedReader, String> {
-            return it.readText()
+    suspend fun getEventString(filePath: String): String =
+        readMutex.withLock {
+            // Block one time of file reads if another task has read the content of this file
+            if (filePathSet.contains(filePath)) {
+                filePathSet.remove(filePath)
+                return@withLock ""
+            }
+            filePathSet.add(filePath)
+            File(filePath).bufferedReader().use<BufferedReader, String> {
+                val content = it.readText()
+                val isCurrentVersion = content.endsWith("\n")
+                if (isCurrentVersion) {
+                    // handle current version
+                    val events = JSONArray()
+                    content.split("\n").forEach {
+                        if (it.isNotEmpty()) {
+                            try {
+                                events.put(JSONObject(it))
+                            } catch (e: JSONException) {
+                                diagnostics.addMalformedEvent(it)
+                                logger.error("Failed to parse event: $it")
+                            }
+                        }
+                    }
+                    return@use if (events.length() > 0) {
+                        events.toString()
+                    } else {
+                        ""
+                    }
+                } else {
+                    // handle earlier versions
+                    val normalizedContent = "[${content.trimStart('[', ',').trimEnd(']', ',')}]"
+                    try {
+                        val jsonArray = JSONArray(normalizedContent)
+                        return@use jsonArray.toString()
+                    } catch (e: JSONException) {
+                        diagnostics.addMalformedEvent(normalizedContent)
+                        logger.error("Failed to parse events: $normalizedContent, dropping file: $filePath")
+                        this.remove(filePath)
+                        return@use ""
+                    }
```

Comment:
Is this block still needed after the migration?

Reply:
In theory it is not needed, I only left this just in case if there is something wrong in migration we still have a backup. Would you prefer we remove this now or it would be better to remove later?

Reply:
Sounds good, I'm ok with this as a safety net for now. Can you maybe add a comment that section is not expected to be called and can be removed later based on diagnostics?

Reply:
Sure, I can add a comment for this.

Reply:
Maybe in the else condition we just send the data as malformed events?

Reply:
@izaaz Just to confirm, would you prefer we return the malformed events or just remove the whole block? In theory this block won't be hit. Though given the edge cases we hit earlier I guess a safety guard won't hurt.

Reply:
I mean we send this data as malformed data to the server so that we can recover if we run into bugs like we did earlier. The problem with safe guards is that we'll start adding it for every new version of the file and this function will soon start getting harder to read.

Reply:
@izaaz I updated to return the whole content if it falls into malformed case, just kept the handling at first. So we could reduce the case for server to handle. Won't worry much about the new version changes as we could always refine and clean up this part.